sdk/core/Azure.Core/tests/TestFramework/RecordSession.cs

```diff
@@ -84,5 +85,36 @@ public void Sanitize(RecordedTestSanitizer sanitizer)
                 }
             }
         }
+
+        public bool IsEquivalent(RecordSession session, RecordMatcher matcher)
+        {
+            if (session == null)
+            {
+                return false;
+            }
+
+            return session.Variables.SequenceEqual(Variables) &&
+                   session.Entries.SequenceEqual(Entries, new EntryEquivalentComparer(matcher));
```

Comment:
When we're doing playback, we scan to find a matching entry but here we're assuming they'll all be in the same order?  I think it's fine for now to see how well that works but we may need to match the existing behavior in the future.

Reply:
I was trying to be slightly strict and seeing if that works, as you've said I'm OK adjusting if we need to.